Tasks/BuildctlV0/src/utils.ts

```diff
@@ -0,0 +1,134 @@
+    if(buildctlVersion && buildctlVersion != "latest") {
+        return sanitizeVersionString(buildctlVersion.trim());
+    }
+    return await getStableBuildctlVersion();
+}
+
+async function getStableBuildctlVersion(): Promise<string> {
+    var request = new webclient.WebRequest();
+    request.uri = buildctlLatestReleaseUrl;
+    request.method = "GET";
+
+    try {
+        var response = await webclient.sendRequest(request);
+        return response.body["tag_name"];
+    } catch (error) {
+        tl.warning(tl.loc("BuildctlLatestNotKnown", buildctlLatestReleaseUrl, error, stableBuildctlVersion));
+    }
+
+    return stableBuildctlVersion;
+}
+export async function downloadBuildctl(version: string): Promise<string> {
+    var cachedToolpath = toolLib.findLocalTool(buildctlToolName, version);
+    let buildctlDownloadPath: string = null;
+    if (!cachedToolpath) {
+        try {
+            buildctlDownloadPath = await toolLib.downloadTool(getBuildctlDownloadURL(version), buildctlToolName + "-" + version + "-" + uuidV4() + getArchiveExtension());
+        } catch (exception) {
+            throw new Error(tl.loc("BuildctlDownloadFailed", getBuildctlDownloadURL(version), exception));
+        }
+
+        var unzipedBuildctlPath = await toolLib.extractTar(buildctlDownloadPath);
+        unzipedBuildctlPath = path.join(unzipedBuildctlPath, "bin", buildctlToolName);
+        tl.debug('Extracting archive: ' + unzipedBuildctlPath+' download path: '+buildctlDownloadPath);
+        var cachedToolpath = await toolLib.cacheFile(unzipedBuildctlPath, buildctlToolName, buildctlToolName, version);
+        tl.debug('CachedTool path: ' + cachedToolpath);    
+    }
+
+    var buildctlpath = findBuildctl(cachedToolpath);
+    if (!buildctlpath) {
+        throw new Error(tl.loc("BuildctlNotFoundInFolder", cachedToolpath))
+    }
+    tl.debug('Buildctl path: ' + buildctlpath);
+    fs.chmodSync(buildctlpath, "777");
+    return buildctlpath;
+}
+
+function getBuildctlDownloadURL(version: string): string {
+    switch (os.type()) {
+        case 'Windows_NT':
+            return util.format("https://github.com/moby/buildkit/releases/download/%s/buildkit-%s.windows-amd64.tar.gz", version, version);
+
+        case 'Darwin':
+            return util.format("https://github.com/moby/buildkit/releases/download/%s/buildkit-%s.darwin-amd64.tar.gz", version, version);
+
+        default:
+            case 'Linux':
+                return util.format("https://github.com/moby/buildkit/releases/download/%s/buildkit-%s.linux-amd64.tar.gz", version, version);
+
+    }
+}
+
+export async function getBuildKitPod(): Promise<string> {
+
+    var consistenthashkey = tl.getVariable('Build.Repository.Name')+tl.getInput("dockerFile", true);
+    var kubectlToolPath = tl.which("kubectl", true);
+    var kubectlTool = tl.tool(kubectlToolPath);
+    kubectlTool.arg('get');
+    kubectlTool.arg('services');
+    kubectlTool.arg('k8s-poolprovider');
```

Comment:
If one changes the values in values.yaml and deploys the chart, this will not work

Reply:
removed hardcoding fetching the value at runtime